tools/scaffolder/internal/generate/config.go

```diff
@@ -15,160 +15,230 @@
-	var mappingsWithConfig []MappingWithConfig
-	for _, mapping := range selectedCRD.Mappings {
-		openAPIConfig, exists := openAPIMap[mapping.OpenAPIRef.Name]
-		if !exists {
-			return nil, fmt.Errorf("OpenAPI config '%s' not found for mapping", mapping.OpenAPIRef.Name)
+	if apiMappingsStr, exists := crd.Metadata.Annotations["api-mappings"]; exists {
+		var apiMapping APIMapping
+		if err := yaml.Unmarshal([]byte(apiMappingsStr), &apiMapping); err != nil {
+			return nil, fmt.Errorf("failed to parse api-mappings annotation: %w", err)
 		}
 
-		mappingsWithConfig = append(mappingsWithConfig, MappingWithConfig{
-			Mapping:       mapping,
-			OpenAPIConfig: openAPIConfig,
+		if specProp, exists := apiMapping.Properties["spec"]; exists {
+			for versionName, versionProp := range specProp.Properties {
+				if versionProp.AtlasSDKVersion != "" {
+					crdInfo.Versions = append(crdInfo.Versions, CRDVersionInfo{
+						Version:         versionName,
+						AtlasSDKVersion: versionProp.AtlasSDKVersion,
+						SDKPackage:      versionProp.AtlasSDKVersion,
+					})
+				}
+			}
+		}
+	}
+
+	if len(crd.Spec.Versions) > 0 {
+		crdInfo.Version = crd.Spec.Versions[0].Name
+	}
+
+	return crdInfo, nil
+}
+
+func ParseCRDConfig(resultPath, crdKind string) (*ParsedConfig, error) {
+	crdInfo, err := ParseCRDResult(resultPath, crdKind)
+	if err != nil {
+		return nil, err
+	}
+
+	var mappings []MappingWithConfig
+	for _, version := range crdInfo.Versions {
+		mappings = append(mappings, MappingWithConfig{
+			Version: version.Version,
+			OpenAPIConfig: OpenAPIConfig{
+				Name:    version.Version,
+				Package: version.SDKPackage,
+			},
 		})
 	}
 
 	return &ParsedConfig{
-		Config:       config,
-		OpenAPIMap:   openAPIMap,
-		CRDMap:       crdMap,
-		SelectedCRD:  selectedCRD,
-		Mappings:     mappingsWithConfig,
-		ResourceName: crdKind,
+		SelectedCRD:  *crdInfo,
+		Mappings:     mappings,
+		ResourceName: crdInfo.Kind,
 	}, nil
 }
 
-// ListCRDs returns a list of all available CRDs from the config file
-func ListCRDs(configPath string) ([]CRDConfig, error) {
-	data, err := os.ReadFile(configPath)
+func ListCRDs(resultPath string) ([]CRDInfo, error) {
+	file, err := os.Open(resultPath)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read config file: %w", err)
+		return nil, fmt.Errorf("failed to open result file: %w", err)
 	}
+	defer file.Close()
+
+	var crds []CRDInfo
+	scanner := bufio.NewScanner(file)
+
+	for {
+		crdInfo, err := parseNextCRD(scanner)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			continue
+		}
 
-	var config Config
-	if err := yaml.Unmarshal(data, &config); err != nil {
-		return nil, fmt.Errorf("failed to parse config YAML: %w", err)
+		if crdInfo != nil {
+			crds = append(crds, *crdInfo)
+		}
 	}
 
-	return config.Spec.CRD, nil
-}
+	if len(crds) == 0 {
+		return nil, fmt.Errorf("no CustomResourceDefinition documents found in '%s' - ensure you're using an openapi2crd result.yaml file", resultPath)
```

Comment:
leave "openapi2crd" out of the picture in the whole code base. This tool just consumes CRD yaml.